From a7cf8dcb18ec77769517ca343b72bd727ab4117b Mon Sep 17 00:00:00 2001
From: Ioana Ciornei <ioana.ciornei@nxp.com>
Date: Fri, 7 Dec 2018 14:15:48 +0530
Subject: [PATCH 396/741] vfio/fsl-mc: Use self-device as consumer to allocate
 mc-portal

commit 669dfeb67eeb5b61ce8dc65d34d9345350ce9fe3 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Earlier fsl_mc_portal_allocate() expect to pass root
dprc container device but now fsl_mc_portal_allocate()
takes care to allocate from root dprc container. Also
fsl_mc_portal_allocate() links allocated mc-portal with
consumer/requester device for binding/unbinding order.
so we need to pass own device as requester.

Signed-off-by: Ioana Ciornei <ioana.ciornei@nxp.com>
Signed-off-by: Bharat Bhushan <Bharat.Bhushan@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/vfio/fsl-mc/vfio_fsl_mc.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/vfio/fsl-mc/vfio_fsl_mc.c b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
index e343c6013429..10738a70f0ae 100644
--- a/drivers/vfio/fsl-mc/vfio_fsl_mc.c
+++ b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
@@ -533,8 +533,7 @@ static int vfio_fsl_mc_initialize_dprc(struct vfio_fsl_mc_device *vdev)
 	if (WARN_ON(!root_dprc_dev))
 		return -EINVAL;
 
-	ret = fsl_mc_portal_allocate(to_fsl_mc_device(root_dprc_dev),
-				     FSL_MC_IO_ATOMIC_CONTEXT_PORTAL,
+	ret = fsl_mc_portal_allocate(mc_dev, FSL_MC_IO_ATOMIC_CONTEXT_PORTAL,
 				     &mc_dev->mc_io);
 	if (ret < 0)
 		goto clean_msi_domain;
-- 
2.17.1

